File: src/main/java/com/ggemo/bilidanmakuclient/structs/SendAuthDO.java
package com.ggemo.bilidanmakuclient.structs;

import com.alibaba.fastjson.annotation.JSONField;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SendAuthDO {
    @JSONField(name="uid")
    private long uId;

    @JSONField(name="roomid")
    private long roomId;

    @JSONField(name="protover")
    private static final int PROTOVER = 2;

    @JSONField(name="platform")
    private static final String PLATFORM = "web";

    @JSONField(name="clientver")
    private static final String CLIENT_VER = "1.8.2";

    @JSONField(name="type")
    private static final int TYPE = 2;

    private String key;
}

File: src/main/java/com/ggemo/va/bilidanmakuclient/structs/SendAuthDO.kt
package com.ggemo.va.bilidanmakuclient.structs

import com.alibaba.fastjson.JSON
import lombok.AllArgsConstructor
import lombok.NoArgsConstructor
import com.alibaba.fastjson.annotation.JSONField
import lombok.Data

class SendAuthDO(uId: Long, roomId: Long, token: String?) {
    @JSONField(name = "uid")
    val uId: Long = uId

    @JSONField(name = "roomid")
    val roomId: Long = roomId
    val key: String? = token

    @JSONField(name = "protover")
    val PROTOVER = 2

    @JSONField(name = "platform")
    val PLATFORM = "web"

    @JSONField(name = "clientver")
    val CLIENT_VER = "1.8.2"

    @JSONField(name = "type")
    val TYPE = 2
}